nemo/utils/exp_manager.py

```diff
@@ -538,19 +542,24 @@ class NeMoModelCheckpoint(ModelCheckpoint):
     """ Light wrapper around Lightning's ModelCheckpoint to force a saved checkpoint on train_end
```

Comment:
In TLT, why are you using `NeMoModelCheckpoint`? Why not just use Lightning's `ModelCheckpoint`? Do you need the final pl_module.save_to() call? Are the "ckpt" saved by Lightning not enough?

Reply:
PTL chkpt 1) doesn't have artifacts (that I need), but 2) contains also trainer/optim params (I do not need).

PTL ckpt is not EFF archive.

Reply:
All `NeMoModelCheckpoint` does is call `save_to` on `on_train_end`.

It DOES NOT save model checkpoints during training. This is delegated to Lightning's callback which saves Lightning's "ckpt"s

> PTL ckpt is not EFF archive.

It is my understanding that you are patching this so that they are.

Reply:
> All NeMoModelCheckpoint does is call save_to on on_train_end.

Exactly - and  in TLT I am relying on  NeMoModelCheckpoint  on_train_end -> calls model save -> calls EFF NeMoCookbook save.